sdk-core/src/test/java/care/data4life/sdk/network/ApiServiceModuleTest.kt

```diff
@@ -0,0 +1,725 @@
+
+        // When
+        val actual = service.uploadDocument(alias, userId, data).blockingGet()
+
+        // Then
+        assertEquals(
+            actual = actual,
+            expected = documentId
+        )
+
+        val request = server.takeRequest()
+
+        assertEquals(
+            actual = request.path,
+            expected = "/users/$userId/documents"
+        )
+        assertNull(request.headers[HEADER_ALIAS])
+        assertEquals(
+            actual = request.headers[HEADER_AUTHORIZATION],
+            expected = "Bearer $authToken"
+        )
+        assertEquals(
+            actual = request.method,
+            expected = "POST"
+        )
+    }
+
+    @Test
+    fun `Given, downloadDocument is called with an Alias, UserId and a DocumentId, it returns a document`() {
+        // Given
+        val alias = ALIAS
+        val userId = USER_ID
+        val documentId = DOCUMENT_ID
+        val authToken = AUTH_TOKEN
+        val payload = "test"
+
+        val response = MockResponse().setBody(payload)
+
+        addResponsesAndStart(listOf(response))
+        simulateAuthService(alias, authToken)
+
+        // When
+        val actual = service.downloadDocument(alias, userId, documentId).blockingGet()
+
+        // Then
+        assertTrue(actual.contentEquals(payload.toByteArray()))
+
+        val request = server.takeRequest()
+
+        assertEquals(
+            actual = request.path,
+            expected = "/users/$userId/documents/$documentId"
+        )
+        assertNull(request.headers[HEADER_ALIAS])
+        assertEquals(
+            actual = request.headers[HEADER_AUTHORIZATION],
+            expected = "Bearer $authToken"
+        )
+        assertEquals(
+            actual = request.method,
+            expected = "GET"
+        )
+    }
+
+    // TODO: FIX this
+    @Test
+    @Ignore("This should work, but it does not, figure out why!")
+    fun `Given, deleteDocument is called with an Alias, UserId and a DocumentId, it returns a completable action`() {
+        // Given
+        val alias = ALIAS
+        val userId = USER_ID
+        val documentId = DOCUMENT_ID
+        val authToken = AUTH_TOKEN
+
+        val response = MockResponse().setResponseCode(204)
+
+        addResponsesAndStart(listOf(response))
+        simulateAuthService(alias, authToken)
+
+        // When
+        val actual = service.deleteDocument(alias, userId, documentId).blockingGet()
+
+        // Then
+        assertTrue(actual)
+
+        val request = server.takeRequest()
+
+        assertEquals(
+            actual = request.path,
+            expected = "/users/$userId/documents/$documentId"
+        )
+        assertNull(request.headers[HEADER_ALIAS])
+        assertEquals(
+            actual = request.headers[HEADER_AUTHORIZATION],
+            expected = "Bearer $authToken"
+        )
+        assertEquals(
+            actual = request.method,
+            expected = "DELETE"
+        )
+    }
```

Comment:
I just want make obvious, that this should work, but it does not. I even not see, why `deleteDocument` has to be different then `deleteRecord` in terms of returning a Completable. Can we align `deleteDocument` and `deleteRecord` so they working similar?